crates/bevy_sprite/src/texture_slice/computed_slices.rs

```diff
@@ -53,10 +53,21 @@ impl ComputedTextureSlices {
                 flip_x,
                 flip_y,
                 image_handle_id: handle.id(),
-                anchor: sprite.anchor.as_vec(),
+                anchor: Self::redepend_anchor_from_sprite_to_slice(sprite, slice),
             }
         })
     }
+
+    fn redepend_anchor_from_sprite_to_slice(sprite: &Sprite, slice: &TextureSlice) -> Vec2 {
+        let sprite_size = sprite
+            .custom_size
+            .unwrap_or(sprite.rect.unwrap_or_default().size());
+        if sprite_size != Vec2::ZERO {
```

Comment:
maybe flip the test around so that we don't have _yoda logic_, i.e. if the `sprite_size == Vec2::Zero` then anchor is unaltered, otherwise....